superset-frontend/src/dashboard/components/nativeFilters/FilterBar/index.tsx

```diff
@@ -238,7 +243,9 @@ const FilterBar: React.FC<FiltersBarProps> = ({
           dataMaskSelected={dataMaskSelected}
           dataMaskApplied={dataMaskApplied}
         />
-        {isFeatureEnabled(FeatureFlag.DASHBOARD_NATIVE_FILTERS_SET) ? (
+        {!isInitialized ? (
+          <Loading />
+        ) : isFeatureEnabled(FeatureFlag.DASHBOARD_NATIVE_FILTERS_SET) ? (
```

Comment:
I think this causes an unnecessary visual artefact - I think it's ok to keep this as-is, i.e. showing the filters in spinning state until they've fully resolved.

Reply:
but this is the point to render them in order they will not start to do requests to the server :)

Reply:
Haha, makes sense 🤦